### PR TITLE
Update Azure.Core version "1.21.0-alpha.20211029.1"

### PR DESCRIPTION
**Pull request by ShivangiReja**
Update Azure.Core `1.21.0-alpha.20211029.1` version in order to prepare release for  LLC WebPubSub.  Following types has been updated:

* RequestOptions -> RequestContext
* ResponseStatusOptions -> ErrorOptions

Before WebPubSub release, we will update the released Azure.Core version.

### CODE REVIEW

File: samples/Azure.AI.DocumentTranslation/Generated/DocumentTranslationClient.cs [hunk 1]
```diff
@@ -61,7 +61,7 @@ public DocumentTranslationClient(string endpoint, AzureKeyCredential credential,
         /// <summary> Returns the translation status for a specific document based on the request Id and document Id. </summary>
         /// <param name="id"> Format - uuid.  The batch id. </param>
         /// <param name="documentId"> Format - uuid.  The document id. </param>
-        /// <param name="options"> The request options. </param>
+        /// <param name="context"> The request options. </param>
```
Comment (annelo-msft) [inline samples/Azure.AI.DocumentTranslation/Generated/DocumentTranslationClient.cs:64]: Would it be possible to update the comment here as well?

File: src/assets/Generator.Shared/HttpPipelineExtensions.cs [hunk 2]
```diff
@@ -97,16 +96,14 @@ public static Response<bool> ProcessHeadAsBoolMessage(this HttpPipeline pipeline
             }
         }
 
-        private static (CancellationToken CancellationToken, ResponseStatusOption StatusOption) ApplyRequestOptions(RequestOptions? requestOptions, HttpMessage message)
+        private static (CancellationToken CancellationToken, ErrorOptions ErrorOptions) ApplyRequestOptions(RequestContext? requestContext, HttpMessage message)
```
Comment (annelo-msft) [inline src/assets/Generator.Shared/HttpPipelineExtensions.cs:99]: nit: could we rename this method to ApplyRequestContext?

**Review by annelo-msft — APPROVED** [review]
Thanks, Shivangi!

File: samples/Azure.AI.DocumentTranslation/Generated/DocumentTranslationClient.cs (hunk 1 above)
Comment (ShivangiReja) [reply samples/Azure.AI.DocumentTranslation/Generated/DocumentTranslationClient.cs]: Updated!

**Pull request merged**